From foo@baz.org Sun Jun  5 14:33:21 2005
Date: Mon, 20 Jun 2005 21:15:16 -0700
To: Greg KH <gregkh@suse.de>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: devfs: Remove the gendisk devfs_name field as it's no longer needed

And remove the now unneeded number field.
Also fixes all drivers that set these fields.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/um/drivers/ubd_kern.c      |    8 ++------
 drivers/block/DAC960.c          |    1 -
 drivers/block/acsi.c            |    5 +----
 drivers/block/cciss.c           |    1 -
 drivers/block/cpqarray.c        |    2 --
 drivers/block/loop.c            |    1 -
 drivers/block/nbd.c             |    1 -
 drivers/block/ps2esdi.c         |    1 -
 drivers/block/rd.c              |    1 -
 drivers/block/swim3.c           |    1 -
 drivers/block/sx8.c             |    1 -
 drivers/block/ub.c              |    2 --
 drivers/block/umem.c            |    1 -
 drivers/block/viodasd.c         |    2 --
 drivers/block/xd.c              |    1 -
 drivers/block/z2ram.c           |    1 -
 drivers/cdrom/aztcd.c           |    1 -
 drivers/cdrom/gscd.c            |    1 -
 drivers/cdrom/optcd.c           |    1 -
 drivers/cdrom/sbpcd.c           |    1 -
 drivers/cdrom/sjcd.c            |    1 -
 drivers/cdrom/sonycd535.c       |    1 -
 drivers/cdrom/viocd.c           |    2 --
 drivers/md/md.c                 |    7 ++-----
 drivers/message/i2o/i2o_block.c |    1 -
 drivers/mmc/mmc_block.c         |    1 -
 drivers/s390/block/dasd_genhd.c |    2 --
 drivers/s390/block/xpram.c      |    1 -
 include/linux/genhd.h           |    2 --
 29 files changed, 5 insertions(+), 47 deletions(-)

--- gregkh-2.6.orig/drivers/block/loop.c
+++ gregkh-2.6/drivers/block/loop.c
@@ -1291,7 +1291,6 @@ static int __init loop_init(void)
 		disk->first_minor = i;
 		disk->fops = &lo_fops;
 		sprintf(disk->disk_name, "loop%d", i);
-		sprintf(disk->devfs_name, "loop/%d", i);
 		disk->private_data = lo;
 		disk->queue = lo->lo_queue;
 	}
--- gregkh-2.6.orig/drivers/block/rd.c
+++ gregkh-2.6/drivers/block/rd.c
@@ -456,7 +456,6 @@ static int __init rd_init(void)
 		disk->queue = rd_queue[i];
 		disk->flags |= GENHD_FL_SUPPRESS_PARTITION_INFO;
 		sprintf(disk->disk_name, "ram%d", i);
-		sprintf(disk->devfs_name, "rd/%d", i);
 		set_capacity(disk, rd_size * 2);
 		add_disk(rd_disks[i]);
 	}
--- gregkh-2.6.orig/include/linux/genhd.h
+++ gregkh-2.6/include/linux/genhd.h
@@ -110,8 +110,6 @@ struct gendisk {
 	sector_t capacity;
 
 	int flags;
-	char devfs_name[64];		/* devfs crap */
-	int number;			/* more of the same */
 	struct device *driverfs_dev;
 	struct kobject kobj;
 
--- gregkh-2.6.orig/drivers/block/ub.c
+++ gregkh-2.6/drivers/block/ub.c
@@ -32,7 +32,6 @@
 #include <scsi/scsi.h>
 
 #define DRV_NAME "ub"
-#define DEVFS_NAME DRV_NAME
 
 #define UB_MAJOR 180
 
@@ -2324,7 +2323,6 @@ static int ub_probe_lun(struct ub_dev *s
 
 	lun->disk = disk;
 	sprintf(disk->disk_name, DRV_NAME "%c", lun->id + 'a');
-	sprintf(disk->devfs_name, DEVFS_NAME "/%c", lun->id + 'a');
 	disk->major = UB_MAJOR;
 	disk->first_minor = lun->id * UB_PARTS_PER_LUN;
 	disk->fops = &ub_bd_fops;
--- gregkh-2.6.orig/drivers/mmc/mmc_block.c
+++ gregkh-2.6/drivers/mmc/mmc_block.c
@@ -401,7 +401,6 @@ static struct mmc_blk_data *mmc_blk_allo
 	 */
 
 	sprintf(md->disk->disk_name, "mmcblk%d", devidx);
-	sprintf(md->disk->devfs_name, "mmc/blk%d", devidx);
 
 	blk_queue_hardsect_size(md->queue.queue, 1 << md->block_bits);
 
--- gregkh-2.6.orig/drivers/cdrom/aztcd.c
+++ gregkh-2.6/drivers/cdrom/aztcd.c
@@ -1918,7 +1918,6 @@ static int __init aztcd_init(void)
 	azt_disk->first_minor = 0;
 	azt_disk->fops = &azt_fops;
 	sprintf(azt_disk->disk_name, "aztcd");
-	sprintf(azt_disk->devfs_name, "aztcd");
 	azt_disk->queue = azt_queue;
 	add_disk(azt_disk);
 	azt_invalidate_buffers();
--- gregkh-2.6.orig/drivers/cdrom/gscd.c
+++ gregkh-2.6/drivers/cdrom/gscd.c
@@ -955,7 +955,6 @@ static int __init gscd_init(void)
 	gscd_disk->first_minor = 0;
 	gscd_disk->fops = &gscd_fops;
 	sprintf(gscd_disk->disk_name, "gscd");
-	sprintf(gscd_disk->devfs_name, "gscd");
 
 	if (register_blkdev(MAJOR_NR, "gscd")) {
 		ret = -EIO;
--- gregkh-2.6.orig/drivers/cdrom/optcd.c
+++ gregkh-2.6/drivers/cdrom/optcd.c
@@ -2033,7 +2033,6 @@ static int __init optcd_init(void)
 	optcd_disk->first_minor = 0;
 	optcd_disk->fops = &opt_fops;
 	sprintf(optcd_disk->disk_name, "optcd");
-	sprintf(optcd_disk->devfs_name, "optcd");
 
 	if (!request_region(optcd_port, 4, "optcd")) {
 		printk(KERN_ERR "optcd: conflict, I/O port 0x%x already used\n",
--- gregkh-2.6.orig/drivers/cdrom/sbpcd.c
+++ gregkh-2.6/drivers/cdrom/sbpcd.c
@@ -5870,7 +5870,6 @@ int __init sbpcd_init(void)
 		disk->fops = &sbpcd_bdops;
 		strcpy(disk->disk_name, sbpcd_infop->name);
 		disk->flags = GENHD_FL_CD;
-		sprintf(disk->devfs_name, "sbp/c0t%d", p->drv_id);
 		p->disk = disk;
 		if (register_cdrom(sbpcd_infop))
 		{
--- gregkh-2.6.orig/drivers/cdrom/sjcd.c
+++ gregkh-2.6/drivers/cdrom/sjcd.c
@@ -1695,7 +1695,6 @@ static int __init sjcd_init(void)
 	sjcd_disk->first_minor = 0,
 	sjcd_disk->fops = &sjcd_fops,
 	sprintf(sjcd_disk->disk_name, "sjcd");
-	sprintf(sjcd_disk->devfs_name, "sjcd");
 
 	if (!request_region(sjcd_base, 4,"sjcd")) {
 		printk
--- gregkh-2.6.orig/drivers/cdrom/sonycd535.c
+++ gregkh-2.6/drivers/cdrom/sonycd535.c
@@ -1589,7 +1589,6 @@ static int __init sony535_init(void)
 	cdu_disk->first_minor = 0;
 	cdu_disk->fops = &cdu_fops;
 	sprintf(cdu_disk->disk_name, "cdu");
-	sprintf(cdu_disk->devfs_name, "cdu535");
 
 	if (!request_region(sony535_cd_base_io, 4, CDU535_HANDLE)) {
 		printk(KERN_WARNING"sonycd535: Unable to request region 0x%x\n",
--- gregkh-2.6.orig/drivers/cdrom/viocd.c
+++ gregkh-2.6/drivers/cdrom/viocd.c
@@ -688,8 +688,6 @@ static int viocd_probe(struct vio_dev *v
 	gendisk->first_minor = deviceno;
 	strncpy(gendisk->disk_name, c->name,
 			sizeof(gendisk->disk_name));
-	snprintf(gendisk->devfs_name, sizeof(gendisk->devfs_name),
-			VIOCD_DEVICE_DEVFS "%d", deviceno);
 	blk_queue_max_hw_segments(q, 1);
 	blk_queue_max_phys_segments(q, 1);
 	blk_queue_max_sectors(q, 4096 / 512);
--- gregkh-2.6.orig/drivers/md/md.c
+++ gregkh-2.6/drivers/md/md.c
@@ -2400,13 +2400,10 @@ static struct kobject *md_probe(dev_t de
 	}
 	disk->major = MAJOR(dev);
 	disk->first_minor = unit << shift;
-	if (partitioned) {
+	if (partitioned)
 		sprintf(disk->disk_name, "md_d%d", unit);
-		sprintf(disk->devfs_name, "md/d%d", unit);
-	} else {
+	else
 		sprintf(disk->disk_name, "md%d", unit);
-		sprintf(disk->devfs_name, "md/%d", unit);
-	}
 	disk->fops = &md_fops;
 	disk->private_data = mddev;
 	disk->queue = mddev->queue;
--- gregkh-2.6.orig/drivers/block/DAC960.c
+++ gregkh-2.6/drivers/block/DAC960.c
@@ -2531,7 +2531,6 @@ static boolean DAC960_RegisterBlockDevic
 	blk_queue_max_sectors(RequestQueue, Controller->MaxBlocksPerCommand);
 	disk->queue = RequestQueue;
 	sprintf(disk->disk_name, "rd/c%dd%d", Controller->ControllerNumber, n);
-	sprintf(disk->devfs_name, "rd/host%d/target%d", Controller->ControllerNumber, n);
 	disk->major = MajorNumber;
 	disk->first_minor = n << DAC960_MaxPartitionsBits;
 	disk->fops = &DAC960_BlockDeviceOperations;
--- gregkh-2.6.orig/drivers/block/acsi.c
+++ gregkh-2.6/drivers/block/acsi.c
@@ -1732,13 +1732,10 @@ int acsi_init( void )
 		struct gendisk *disk = acsi_gendisk[i];
 		sprintf(disk->disk_name, "ad%c", 'a'+i);
 		aip = &acsi_info[NDevices];
-		sprintf(disk->devfs_name, "ad/target%d/lun%d", aip->target, aip->lun);
 		disk->major = ACSI_MAJOR;
 		disk->first_minor = i << 4;
-		if (acsi_info[i].type != HARDDISK) {
+		if (acsi_info[i].type != HARDDISK)
 			disk->minors = 1;
-			strcat(disk->devfs_name, "/disc");
-		}
 		disk->fops = &acsi_fops;
 		disk->private_data = &acsi_info[i];
 		set_capacity(disk, acsi_info[i].size);
--- gregkh-2.6.orig/drivers/block/cciss.c
+++ gregkh-2.6/drivers/block/cciss.c
@@ -3232,7 +3232,6 @@ static int __devinit cciss_init_one(stru
 
 		q->queuedata = hba[i];
 		sprintf(disk->disk_name, "cciss/c%dd%d", i, j);
-		sprintf(disk->devfs_name, "cciss/host%d/target%d", i, j);
 		disk->major = hba[i]->major;
 		disk->first_minor = j << NWD_SHIFT;
 		disk->fops = &cciss_fops;
--- gregkh-2.6.orig/drivers/block/cpqarray.c
+++ gregkh-2.6/drivers/block/cpqarray.c
@@ -1805,8 +1805,6 @@ static void getgeometry(int ctlr)
 
 				}
 
-				sprintf(disk->devfs_name, "ida/c%dd%d", ctlr, log_unit);
-
 				info_p->phys_drives =
 				    sense_config_buf->ctlr_phys_drv;
 				info_p->drv_assign_map
--- gregkh-2.6.orig/drivers/block/nbd.c
+++ gregkh-2.6/drivers/block/nbd.c
@@ -690,7 +690,6 @@ static int __init nbd_init(void)
 		disk->private_data = &nbd_dev[i];
 		disk->flags |= GENHD_FL_SUPPRESS_PARTITION_INFO;
 		sprintf(disk->disk_name, "nbd%d", i);
-		sprintf(disk->devfs_name, "nbd/%d", i);
 		set_capacity(disk, 0x7ffffc00ULL << 1); /* 2 TB */
 		add_disk(disk);
 	}
--- gregkh-2.6.orig/drivers/block/ps2esdi.c
+++ gregkh-2.6/drivers/block/ps2esdi.c
@@ -421,7 +421,6 @@ static int __init ps2esdi_geninit(void)
 		disk->major = PS2ESDI_MAJOR;
 		disk->first_minor = i<<6;
 		sprintf(disk->disk_name, "ed%c", 'a'+i);
-		sprintf(disk->devfs_name, "ed/target%d", i);
 		disk->fops = &ps2esdi_fops;
 		ps2esdi_gendisk[i] = disk;
 	}
--- gregkh-2.6.orig/drivers/block/swim3.c
+++ gregkh-2.6/drivers/block/swim3.c
@@ -1061,7 +1061,6 @@ int swim3_init(void)
 		disk->queue = swim3_queue;
 		disk->flags |= GENHD_FL_REMOVABLE;
 		sprintf(disk->disk_name, "fd%d", i);
-		sprintf(disk->devfs_name, "floppy/%d", i);
 		set_capacity(disk, 2880);
 		add_disk(disk);
 	}
--- gregkh-2.6.orig/drivers/block/sx8.c
+++ gregkh-2.6/drivers/block/sx8.c
@@ -1509,7 +1509,6 @@ static int carm_init_disks(struct carm_h
 		port->disk = disk;
 		sprintf(disk->disk_name, DRV_NAME "/%u",
 			(unsigned int) (host->id * CARM_MAX_PORTS) + i);
-		sprintf(disk->devfs_name, DRV_NAME "/%u_%u", host->id, i);
 		disk->major = host->major;
 		disk->first_minor = i * CARM_MINORS_PER_MAJOR;
 		disk->fops = &carm_bd_ops;
--- gregkh-2.6.orig/drivers/block/umem.c
+++ gregkh-2.6/drivers/block/umem.c
@@ -1191,7 +1191,6 @@ static int __init mm_init(void)
 	for (i = 0; i < num_cards; i++) {
 		struct gendisk *disk = mm_gendisk[i];
 		sprintf(disk->disk_name, "umem%c", 'a'+i);
-		sprintf(disk->devfs_name, "umem/card%d", i);
 		spin_lock_init(&cards[i].lock);
 		disk->major = major_nr;
 		disk->first_minor  = i << MM_SHIFT;
--- gregkh-2.6.orig/drivers/block/viodasd.c
+++ gregkh-2.6/drivers/block/viodasd.c
@@ -523,8 +523,6 @@ retry:
 	else
 		snprintf(g->disk_name, sizeof(g->disk_name),
 				VIOD_GENHD_NAME "%c", 'a' + (dev_no % 26));
-	snprintf(g->devfs_name, sizeof(g->devfs_name),
-			"%s%d", VIOD_GENHD_DEVFS_NAME, dev_no);
 	g->fops = &viodasd_fops;
 	g->queue = q;
 	g->private_data = d;
--- gregkh-2.6.orig/drivers/block/xd.c
+++ gregkh-2.6/drivers/block/xd.c
@@ -215,7 +215,6 @@ static int __init xd_init(void)
 		disk->major = XT_DISK_MAJOR;
 		disk->first_minor = i<<6;
 		sprintf(disk->disk_name, "xd%c", i+'a');
-		sprintf(disk->devfs_name, "xd/target%d", i);
 		disk->fops = &xd_fops;
 		disk->private_data = p;
 		disk->queue = xd_queue;
--- gregkh-2.6.orig/drivers/block/z2ram.c
+++ gregkh-2.6/drivers/block/z2ram.c
@@ -354,7 +354,6 @@ z2_init(void)
     z2ram_gendisk->first_minor = 0;
     z2ram_gendisk->fops = &z2_fops;
     sprintf(z2ram_gendisk->disk_name, "z2ram");
-    strcpy(z2ram_gendisk->devfs_name, z2ram_gendisk->disk_name);
 
     z2ram_gendisk->queue = z2_queue;
     add_disk(z2ram_gendisk);
--- gregkh-2.6.orig/arch/um/drivers/ubd_kern.c
+++ gregkh-2.6/arch/um/drivers/ubd_kern.c
@@ -631,14 +631,10 @@ static int ubd_new_disk(int major, u64 s
 	disk->first_minor = unit << UBD_SHIFT;
 	disk->fops = &ubd_blops;
 	set_capacity(disk, size / 512);
-	if(major == MAJOR_NR){
+	if(major == MAJOR_NR)
 		sprintf(disk->disk_name, "ubd%c", 'a' + unit);
-		sprintf(disk->devfs_name, "ubd/disc%d", unit);
-	}
-	else {
+	else
 		sprintf(disk->disk_name, "ubd_fake%d", unit);
-		sprintf(disk->devfs_name, "ubd_fake/disc%d", unit);
-	}
 
 	/* sysfs register (not for ide fake devices) */
 	if (major == MAJOR_NR) {
--- gregkh-2.6.orig/drivers/message/i2o/i2o_block.c
+++ gregkh-2.6/drivers/message/i2o/i2o_block.c
@@ -1089,7 +1089,6 @@ static int i2o_block_probe(struct device
 	gd = i2o_blk_dev->gd;
 	gd->first_minor = unit << 4;
 	sprintf(gd->disk_name, "i2o/hd%c", 'a' + unit);
-	sprintf(gd->devfs_name, "i2o/hd%c", 'a' + unit);
 	gd->driverfs_dev = &i2o_dev->device;
 
 	/* setup request queue */
--- gregkh-2.6.orig/drivers/s390/block/dasd_genhd.c
+++ gregkh-2.6/drivers/s390/block/dasd_genhd.c
@@ -68,8 +68,6 @@ dasd_gendisk_alloc(struct dasd_device *d
 	}
 	len += sprintf(gdp->disk_name + len, "%c", 'a'+(device->devindex%26));
 
- 	sprintf(gdp->devfs_name, "dasd/%s", device->cdev->dev.bus_id);
-
 	if (device->features & DASD_FEATURE_READONLY)
 		set_disk_ro(gdp, 1);
 	gdp->private_data = device;
--- gregkh-2.6.orig/drivers/s390/block/xpram.c
+++ gregkh-2.6/drivers/s390/block/xpram.c
@@ -466,7 +466,6 @@ static int __init xpram_setup_blkdev(voi
 		disk->private_data = &xpram_devices[i];
 		disk->queue = xpram_queue;
 		sprintf(disk->disk_name, "slram%d", i);
-		sprintf(disk->devfs_name, "slram/%d", i);
 		set_capacity(disk, xpram_sizes[i] << 1);
 		add_disk(disk);
 	}
